torch/csrc/autograd/custom_function.cpp

```diff
@@ -332,7 +332,7 @@ optional_variable_list _process_backward_mode_ad(
       var.mutable_grad().reset();
       impl::clear_hooks(var);
       if (auto grad_acc_fn = impl::try_get_grad_accumulator(var)) {
-        auto grad_acc = dynamic_cast<AccumulateGrad*>(grad_acc_fn.get());
+        auto grad_acc = std::dynamic_pointer_cast<AccumulateGrad>(std::move(grad_acc_fn));
```

Comment:
@cyyever if you read the full comment chain, this doesn't actually work until C++20 so it will still be a bit of perf regression over the code that was previously there. Can you revert the changes to this file?

Reply:
Or if you want to change it, it should be `auto *grad_acc =` ...

Reply:
Can we discard the move? I want to use dynamic_pointer_cast to avoid dereference a possible nullptr